### ISSUE TITLE
this.userId not set when collection updated

### ISSUE THREAD

**Issue by ghost** [maximummeteor/server-transform#9]
Hello,

I am using the collection hooks package in conjunction with my publish functions.  It works great the first time it runs. However, when the collection is modified, and the publishTransformed function reruns,  `this.userId` is no longer set, so my publish function doesn't work correctly.

Here is the offending code from my publishTransformed function

```
    var results = Leads.find(searchCriteria,safeOptions);
        results.serverTransform({

            personTransformed: function(doc) {
            console.log('USERID IS ',this.userId); //THIS LINE PRINTS `UNDEFINED` when the collection is updates
                return Entities.findOne({
                    _id: doc.personId
                },{fields:{firstName:1,lastName:1}});
            },
```

Do you know why this might be?  I can try to make a test case if that would help.

thanks!

**Comment by maxnowack** [comment]
Yes, `this` inside a transformation should be in the context of the publication itself. I'll fix that in a couple of days.
In the meantime, you can use `bind` as a workaround (or maybe a better and independent solution):

``` javascript
    personTransformed: (function(doc) {
        ...
    }).bind(this)
```

**Comment by ghost** [comment]
Ok, I will try that.  Thank you.
